src/main/app.js

```diff
@@ -584,6 +593,28 @@ gpii.app.handleSessionStop = function (that, keyedOutUserToken) {
     }
 };
 
+/**
+ * Handles the onMessage event of the gpii.windows.messages component.
+ *
+ * @param {Component} that An instance of gpii.app
+ * @param {Number} hwnd Window handle.
+ * @param {Number} msg The message.
+ * @param {Number} wParam Message parameter.
+ * @param {Number} lParam Message parameter.
+ * @param {Object} result Set a 'value' field to specify a return value.
+ */
+gpii.app.windowMessage = function (that, hwnd, msg, wParam, lParam, result) {
+    console.log(hwnd, msg, wParam, lParam, result);
+    // https://msdn.microsoft.com/library/aa376889
+    var WM_QUERYENDSESSION = 0x11;
+    if (msg === WM_QUERYENDSESSION) {
+        console.log("SHUTDOWN");
```

Comment:
Add a little bit more context here since short generic messages dumped to the console can be frustrating for people trying to track down where they come from